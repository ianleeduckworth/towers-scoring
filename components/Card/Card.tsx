import * as React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { defaults, globalStyles } from '../../styles/global';

interface CardProps {
    gameName: string;
    date: Date;
    onGamePress: () => void;
    onExitPress: () => void;
}

export const Card = ({ gameName, date, onGamePress: onGamePress, onExitPress }: CardProps) => {
    return (
        <View style={styles.container}>
            <TouchableOpacity style={styles.infoContainer} onPress={onGamePress}>
                <Text style={{...styles.text, marginBottom: 10}}>{gameName}</Text>
                <Text style={styles.text}>{date.toLocaleDateString()}</Text>
            </TouchableOpacity>
            <TouchableOpacity style={styles.exitContainer} onPress={onExitPress}>
                <Text style={styles.text}>x</Text>
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        minWidth: '100%',
        height: 100,
        backgroundColor: defaults.lightGray,
        paddingLeft: 20,
        paddingRight: 20,
        flexDirection: 'row',
        borderWidth: 1,
        borderTopColor: defaults.black,
        borderBottomColor: defaults.black,
    },
    infoContainer: {
        flex: 1,
        justifyContent: 'center',
    },
    text: {
        ...globalStyles.paragraphText
    },
    exitContainer: {
        justifyContent: 'center',
    },
})