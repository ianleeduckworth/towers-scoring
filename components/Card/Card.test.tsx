import renderer from 'react-test-renderer';
import * as React from 'react';
import { Card } from './Card';
import { noOp } from '../../utilities/test/testUtilities';

describe('card', () => {
  it('renders correctly', () => {
    const element = renderer.create(
            <Card
                date={new Date('12/23/2020')}
                gameName={'Test Game Name'}
                onExitPress={noOp}
                onGamePress={noOp}
            />,
    ).toJSON();

    expect(element).toMatchSnapshot();
  });
});
