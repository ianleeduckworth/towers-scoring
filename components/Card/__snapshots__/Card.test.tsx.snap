// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8F8",
      "borderBottomColor": "#000000",
      "borderTopColor": "#000000",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 100,
      "minWidth": "100%",
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      Test Game Name
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      12/23/2020
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      x
    </Text>
  </View>
</View>
`;
